### DISCUSSION TITLE
How can I add opacity property to Run(for animation)?

### DISCUSSION THREAD

**Discussion by kyuranger** [AvaloniaUI/Avalonia#15882] [Q&A] [answered]
I want to add some Run control to TextBlock and then set its opacity to specified value.

However, I found that Run does not have an Opacity property.

Then I tried to add another control (such as border) as its parent and found it is not a control that cannot become as child as any control.

I found the source code of any other control and it seems Opacity property must be inherited from Layoutable.

Run is inherited from Inline and I can't mix Layoutable and Inline together.

Although I could use a textblock as an inline element instead, I feel that textblock is too cumbersome compared to run.

How can I solve my problem? Thank you.

**Comment by rabbitism** [comment]
Just use a forrground with lower alpha value.

**Reply by kyuranger** [reply]
Actually what I need is an opacity animation while not only a opacity color.

**Reply by stevemonaco** [reply, marked as answer]
You could still use a `BrushTransition`.

`BuildTextRun` is `internal`, so it might be hard to implement `Opacity` with your own custom `Inline` element? Not really sure how `Inline` rendering works.

**Reply by Gillibald** [reply]
Only InlineUIContainer will perform well because it is rendering independently

**Reply by kyuranger** [reply]
BrushTransition works! Thank you.
